Sources/PenguinParallel/NonblockingThreadPool/NonBlockingThreadPool.swift

```diff
@@ -462,35 +466,40 @@ fileprivate final class PerThreadState<Environment: ConcurrencyPlatform> {
   // possible to provide a safer API that doesn't leak by default without inducing an extra pointer
   // dereference on critical paths. :-(
 
+  let totalThreadCount: Int
+  let coprimes: [Int]
+  let queues: [NonBlockingThreadPool<Environment>.Queue]
+  let condition: NonblockingCondition<Environment>
+
   var rng: PCGRandomNumberGenerator
 
   var isCancelled: Bool { pool.cancelled }
 
   func steal() -> Task? {
     let r = Int(rng.next())
-    var selectedThreadId = fastFit(r, into: pool.totalThreadCount)
-    let step = pool.coprimes[fastFit(r, into: pool.coprimes.count)]
+    var selectedThreadId = fastFit(r, into: totalThreadCount)
+    let step = coprimes[fastFit(r, into: coprimes.count)]
     assert(
-      step < pool.totalThreadCount, "step: \(step), pool threadcount: \(pool.totalThreadCount)")
+      step < totalThreadCount, "step: \(step), pool threadcount: \(totalThreadCount)")
 
-    for i in 0..<pool.totalThreadCount {
+    for i in 0..<totalThreadCount {
       assert(
-        selectedThreadId < pool.totalThreadCount,
-        "\(selectedThreadId) is too big on iteration \(i); max: \(pool.totalThreadCount), step: \(step)"
+        selectedThreadId < totalThreadCount,
+        "\(selectedThreadId) is too big on iteration \(i); max: \(totalThreadCount), step: \(step)"
       )
-      if let task = pool.queues[selectedThreadId].popBack() {
+      if let task = queues[selectedThreadId].popBack() {
         return task
       }
       selectedThreadId += step
-      if selectedThreadId >= pool.totalThreadCount {
-        selectedThreadId -= pool.totalThreadCount
+      if selectedThreadId >= totalThreadCount {
+        selectedThreadId -= totalThreadCount
       }
     }
     return nil
   }
 
   func spin() -> Task? {
-    let spinCount = pool.threads.count > 0 ? Constants.spinCount / pool.threads.count : 0
+    let spinCount = totalThreadCount > 0 ? Constants.spinCount / totalThreadCount : 0
```

Comment:
This change looks suspect.  Elsewhere you have replaced `pool.totalThreadCount` with `totalThreadCount` but here you're replacing `pool.threads.count` which is presumably a different quantity (if it isn't, I question the value of `pool.totalThreadCount`!)

Reply:
Good eye! This should be the number of worker threads and not the `totalThreadCount`. Thanks!